api/src/main/java/org/apache/iceberg/IncrementalChangelogScan.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+/**
+ * API for configuring a scan for table changes.
+ */
+public interface IncrementalChangelogScan extends Scan<IncrementalChangelogScan> {
```

Comment:
@flyrain @stevenzwu, there was a question whether we should have common hierarchy for this scan and the incremental append scan. I am not sure it is a good idea as those two are fairly different. They are both incremental, though. We could consider having `IncrementalScan` but then `IncrementalAppendScan` is going to be empty.

Reply:
That's a good point. The interface `IncrementalAppendScan` doesn't have any method specified for `Append`. How about replacing `IncrementalAppendScan` with `IncrementalScan`, so that `IncrementalChangelogScan` can extend `IncrementalScan`? And we can have `BaseIncrementalAppendScan` implements `IncrementalScan` directly.

Reply:
> We could consider having IncrementalScan but then IncrementalAppendScan is going to be empty

That was our original plan too. @rdblue was preferring to that we do the refactoring when we actually come to the incremental changelog scan interface. so this PR might be a good place to make the change.

Reply:
@stevenzwu, was the plan to not have  `IncrementalAppendScan ` at all and return `IncrementalScan` from `newIncrementalAppendScan` in `Table`? Do we anticipate any methods specific to append-only scans? Like whether to ignore overwrite snapshots or fail if an overwrite is present?

Reply:
We can extract an `IncrementalScan` parent interface so that `IncrementalAppendScan` and `IncrementalChangelogScan` can extends from. Right now, we don't anticipate any methods specific to append-only scans. We should still defines/keeps the `IncrementalAppendScan` which is identical to `IncrementalScan` (at least for now). 

`Table#newIncrementalAppendScan` should return `IncrementalAppendScan`. In the future, `Table#newIncrementalChangelogScan` can return the `IncrementalChangelogScan` defined in this PR.